### PR TITLE
Fix timestamp_add_interval_months to pass any date

### LINKED ISSUE #6814: Fix timestamp_add_interval_months to pass on any date
### Describe the bug
Ref #3321 

timestamp_add_interval_months fails since month is 7, result is 0.

https://github.com/apache/arrow-datafusion/blob/84832ac3f13af78336f3fa291e88e05f3909faac/datafusion/core/tests/sql/timestamp.rs#L583-L586

### To Reproduce

Test this in July.

### Expected behavior

Pass the test whenever the date we run it.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jayzhan211**
# Which issue does this PR close?

Closes #6814.

# Rationale for this change

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/core/tests/sql/timestamp.rs [hunk 1]
```diff
@@ -579,8 +579,8 @@ async fn timestamp_add_interval_months() -> Result<()> {
     let t1_naive = chrono::NaiveDateTime::parse_from_str(res1, format).unwrap();
     let t2_naive = chrono::NaiveDateTime::parse_from_str(res2, format).unwrap();
 
-    let year = t1_naive.year() + (t1_naive.month() as i32 + 17) / 12;
-    let month = (t1_naive.month() + 17) % 12;
+    let year = t1_naive.year() + (t1_naive.month0() as i32 + 17) / 12;
```
Comment (Dandandan) [inline datafusion/core/tests/sql/timestamp.rs:582]: Instead of using `NOW()` we could use some table of dates instead to test it?.

Comment (Dandandan) [reply datafusion/core/tests/sql/timestamp.rs:582]: This would avoid the test breaking suddenly on a certain date.

**Review by Dandandan — APPROVED** [review]
Approve to get the main branch passing. Left a comment to improve the test

File: datafusion/core/tests/sql/timestamp.rs (hunk 1 above)
Comment (jayzhan211) [reply datafusion/core/tests/sql/timestamp.rs:582]: @Dandandan It seems your suggestion is great, it failed today again :(

**Pull request merged**